Incompatibility with Creeperface01's GAC

# 💬 Let's talk
<!--✍ Feel free to ask questions or start related discussion below -->
pls help  me


### 📋 Debug information
<!-- ⚠ This information may help us to give you better answers but they are not required ⚠ -->
<!-- Use the 'debugpaste' command in PowerNukkit -->
<!-- You can get the version from the file name, the 'about' or 'debugpaste' command outputs -->
* PowerNukkit version: 1.3.0.0-PN
* YouTrack: [PN-123](https://youtrack.gamemods.com.br/issue/PN-123)
* Debug link: https://hastebin.com/ofotohayah

### 💬 Anything else we should know?
<!-- ✍ This is the perfect place to add any additional details -->
```java
22:01:54 [ERROR] Throwing
java.lang.NoSuchMethodException: cz.creeperface.nukkit.gac.player.NukkitCheatPlayer.<init>(cn.nukkit.network.SourceInterface, java.lang.Long, java.net.InetSocketAddress)
        at java.lang.Class.getConstructor0(Class.java:3082) ~[?:1.8.0_231]
        at java.lang.Class.getConstructor(Class.java:1825) ~[?:1.8.0_231]
        at cn.nukkit.network.RakNetInterface.onSessionCreation(RakNetInterface.java:209) [nukkit-1.0-SNAPSHOT.jar:?]
        at com.nukkitx.network.raknet.RakNetServer.onOpenConnectionRequest1(RakNetServer.java:171) [nukkit-1.0-SNAPSHOT.jar:?]
        at com.nukkitx.network.raknet.RakNetServer.access$300(RakNetServer.java:31) [nukkit-1.0-SNAPSHOT.jar:?]
        at com.nukkitx.network.raknet.RakNetServer$ServerDatagramHandler.channelRead(RakNetServer.java:280) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1422) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:931) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.epoll.EpollDatagramChannel.read(EpollDatagramChannel.java:688) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.epoll.EpollDatagramChannel.access$100(EpollDatagramChannel.java:57) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.epoll.EpollDatagramChannel$EpollDatagramChannelUnsafe.epollInReady(EpollDatagramChannel.java:507) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:502) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1050) [nukkit-1.0-SNAPSHOT.jar:?]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [nukkit-1.0-SNAPSHOT.jar:?]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_231]
```

i onley edit the jar name for my start.sh

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -324,6 +324,7 @@ Fixes several anvil issues.
 [updated changelog]:https://github.com/GameModsBR/PowerNukkit/blob/bleeding/CHANGELOG.md
 
 [Unreleased 1.3.1.0-PN]: https://github.com/GameModsBR/PowerNukkit/compare/v1.3.0.0-PN...bleeding
+[Unreleased 1.3.0.1-PN]: https://github.com/GameModsBR/PowerNukkit/compare/v1.3.0.0-PN...bleeding
 [1.3.0.0-PN]: https://github.com/GameModsBR/PowerNukkit/compare/v1.2.1.0-PN...v1.3.0.1-PN
 [1.3.0.0-PN]: https://github.com/GameModsBR/PowerNukkit/compare/v1.2.1.0-PN...v1.3.0.0-PN
 [1.2.1.0-PN]: https://github.com/GameModsBR/PowerNukkit/compare/v1.2.0.2-PN...v1.2.1.0-PN

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@ Click the link above to see the future.
 
 ### Fixes
 - [#320] Multiple output crafting, cake for example
+- [#323] Compatibility issue with the regular version of GAC
 
 ## [1.3.0.0-PN] - 2020-07-01 ([Check the milestone](https://github.com/GameModsBR/PowerNukkit/milestone/11?closed=1))
 Added support for Bedrock Edition 1.16.0 and 1.16.1
@@ -323,6 +324,7 @@ Fixes several anvil issues.
 [updated changelog]:https://github.com/GameModsBR/PowerNukkit/blob/bleeding/CHANGELOG.md
 
 [Unreleased 1.3.1.0-PN]: https://github.com/GameModsBR/PowerNukkit/compare/v1.3.0.0-PN...bleeding
+[1.3.0.0-PN]: https://github.com/GameModsBR/PowerNukkit/compare/v1.2.1.0-PN...v1.3.0.1-PN
 [1.3.0.0-PN]: https://github.com/GameModsBR/PowerNukkit/compare/v1.2.1.0-PN...v1.3.0.0-PN
 [1.2.1.0-PN]: https://github.com/GameModsBR/PowerNukkit/compare/v1.2.0.2-PN...v1.2.1.0-PN
 [1.2.0.2-PN]: https://github.com/GameModsBR/PowerNukkit/compare/v1.2.0.1-PN...v1.2.0.2-PN
@@ -407,3 +409,4 @@ Fixes several anvil issues.
 [#285]: https://github.com/GameModsBR/PowerNukkit/pull/285
 [#293]: https://github.com/GameModsBR/PowerNukkit/pull/293
 [#320]: https://github.com/GameModsBR/PowerNukkit/pull/320
+[#323]: https://github.com/GameModsBR/PowerNukkit/issues/323

--- a/src/main/java/cn/nukkit/Player.java
+++ b/src/main/java/cn/nukkit/Player.java
@@ -1,6 +1,7 @@
 package cn.nukkit;
 
 import cn.nukkit.AdventureSettings.Type;
+import cn.nukkit.api.PowerNukkitOnly;
 import cn.nukkit.block.*;
 import cn.nukkit.blockentity.BlockEntity;
 import cn.nukkit.blockentity.BlockEntityItemFrame;
@@ -28,7 +29,6 @@ import cn.nukkit.event.player.*;
 import cn.nukkit.event.player.PlayerAsyncPreLoginEvent.LoginResult;
 import cn.nukkit.event.player.PlayerInteractEvent.Action;
 import cn.nukkit.event.player.PlayerTeleportEvent.TeleportCause;
-import cn.nukkit.event.player.PlayerJumpEvent;
 import cn.nukkit.event.server.DataPacketReceiveEvent;
 import cn.nukkit.event.server.DataPacketSendEvent;
 import cn.nukkit.form.window.FormWindow;
@@ -83,7 +83,9 @@ import lombok.extern.log4j.Log4j2;
 import java.io.File;
 import java.io.IOException;
 import java.lang.reflect.Field;
+import java.net.InetAddress;
 import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
 import java.nio.ByteOrder;
 import java.util.*;
 import java.util.Map.Entry;
@@ -607,7 +609,20 @@ public class Player extends EntityHuman implements CommandSender, InventoryHolde
     public Map<String, PermissionAttachmentInfo> getEffectivePermissions() {
         return this.perm.getEffectivePermissions();
     }
-
+    
+    private static InetSocketAddress uncheckedNewInetSocketAddress(String ip, int port) {
+        try {
+            return new InetSocketAddress(InetAddress.getByName(ip), port);
+        } catch (UnknownHostException exception) {
+            throw new IllegalArgumentException(exception);
+        }
+    }
+    
+    public Player(SourceInterface interfaz, Long clientID, String ip, int port) {
+        this(interfaz, clientID, uncheckedNewInetSocketAddress(ip, port));
+    }
+    
+    @PowerNukkitOnly
     public Player(SourceInterface interfaz, Long clientID, InetSocketAddress socketAddress) {
         super(null, new CompoundTag());
         this.interfaz = interfaz;

--- a/src/main/java/cn/nukkit/network/RakNetInterface.java
+++ b/src/main/java/cn/nukkit/network/RakNetInterface.java
@@ -19,7 +19,6 @@ import lombok.RequiredArgsConstructor;
 import lombok.extern.log4j.Log4j2;
 
 import java.lang.reflect.Constructor;
-import java.lang.reflect.InvocationTargetException;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
 import java.nio.charset.StandardCharsets;
@@ -205,16 +204,27 @@ public class RakNetInterface implements RakNetServerListener, AdvancedSourceInte
         this.server.getPluginManager().callEvent(ev);
         Class<? extends Player> clazz = ev.getPlayerClass();
 
+        Player player;
+        InetSocketAddress socketAddress = ev.getSocketAddress();
         try {
-            Constructor constructor = clazz.getConstructor(SourceInterface.class, Long.class, InetSocketAddress.class);
-            Player player = (Player) constructor.newInstance(this, ev.getClientId(), ev.getSocketAddress());
-            this.server.addPlayer(session.getAddress(), player);
-            NukkitSessionListener listener = new NukkitSessionListener(player);
-            this.sessionListeners.add(listener);
-            session.setListener(listener);
-        } catch (NoSuchMethodException | InvocationTargetException | InstantiationException | IllegalAccessException e) {
-            Server.getInstance().getLogger().logException(e);
+            Constructor<? extends Player> constructor = clazz.getConstructor(SourceInterface.class, Long.class, InetSocketAddress.class);
+            player = constructor.newInstance(this, ev.getClientId(), socketAddress);
+        } catch (ReflectiveOperationException e) {
+            try {
+                Constructor<? extends Player> constructor = clazz.getConstructor(SourceInterface.class, Long.class, String.class, Integer.TYPE);
+                player = constructor.newInstance(this, ev.getClientId(), socketAddress.getHostString(), socketAddress.getPort());
+            } catch (ReflectiveOperationException e2) {
+                e2.addSuppressed(e);
+                Server.getInstance().getLogger().logException(e);
+                session.disconnect();
+                return;
+            }
         }
+        
+        this.server.addPlayer(session.getAddress(), player);
+        NukkitSessionListener listener = new NukkitSessionListener(player);
+        this.sessionListeners.add(listener);
+        session.setListener(listener);
     }
 
     @Override

